File: client/app/domain/message.ts
export class Message {
  room: string;
  from: string;
  timestamp: number;
  body: string;
  type: string;

  constructor(roomId: string, from: string, body: string, timestamp: number, type?: string) {
    this.room = roomId;
    this.from = from;
    this.body = body;
    this.timestamp = timestamp;
    this.type = type || 'message';
  }
}

File: models/message.go
package models

type Message struct {
  ID        string            `json:"id" bson:"_id"`
  RoomID    string            `json:"room" bson:"room"`
  FromID    string            `json:"from" bson:"from"`
  Timestamp Timestamp         `json:"timestamp" bson:"timestamp"`
  Type      string            `json:"type" bson:"-"`
  Body      string            `json:"body" bson:"body"`
  Reactions map[string]string `json:"reactions,omitempty" bson:"reactions,omitempty"`
}
